src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/openai/OpenAIRequestBody.kt

```diff
@@ -1,9 +1,30 @@
 package org.jetbrains.research.testspark.tools.llm.generation.openai
 
-import org.jetbrains.research.testspark.core.data.ChatMessage
+/**
+ * Adheres the naming of fields for OpenAI chat completion API and checks the correctness of a `role`.
+ * <br/>
+ * Use this class as a carrier of messages that should be sent to OpenAI API.
+ */
+data class OpenAIChatMessage(val role: String, val content: String) {
+    private companion object {
+        /**
+         * The API strictly defines the set of roles.
+         * The `function` role is omitted because it is already deprecated.
+         *
+         * See: https://platform.openai.com/docs/api-reference/chat/create
+         */
+        val supportedRoles = listOf("user", "assistant", "system", "tool")
+    }
+
+    init {
+        if (!supportedRoles.contains(role)) {
+            throw IllegalArgumentException("'$role' is not supported ${OpenAIChatMessage::class}. Available roles are: ${(supportedRoles.joinToString(", ") { "'$it'" })}")
```

Comment:
**This exception may never be thrown unless it is a mistake of a programmer.** 

The `role` is something we programmically install in the code for every `ChatMessage`; this `if` validates that a programmer did not make a silly mistake by confusing string representations of roles that OpenAI expects to see in the JSON (e.g., `"User"` role instead of the correct `"user"`; this is the exact mistake I did when refactored the `ChatMessage` class before).